hnn_core/neuron.py

```diff
@@ -425,35 +425,93 @@ def _create_cells_and_feeds(self):
             else:
                 _PC.cell(gid, feed_cell.nrn_netcon)
 
+    def _connect_celltypes(self, src_type, target_type, loc,
+                           receptor, nc_dict):
```

Comment:
I think we are getting closer to being able to pass a connectivity matrix ...

ideas on a good/flexible/high-level API are welcome. This was just a first pass. Maybe this could also become a public function if users want to make their own connections.

Reply:
So the idea here is to make one method of `Network` that calls `parconnect_from_src` and then explicitly define the connections in `Network._parnet_connect()`?

Reply:
I really like where this is going btw!

Reply:
yes, the connectivity structure in the old code was really obscure to me. I think it should be transparent and easy to change :) Thanks for the excellent reviews, #162 would not have been possible if we didn't add `sect_loc` attribute also to Basket class and make the calls to `_connect_feed_at_loc` consistent between Pyramidal and Basket class in the previous PR.

Reply:
`Network._parnet_connect` is the default connectivity that HNN provides but I'm imagining that the user will be able to modify it easily if they want.